### PR TITLE
Docs: Implement and correct the React examples returned from the unfinished transformer.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
**IMPORTANT: This branch is based on `feature/react-docs-develop`, as it contains the target structure of the documentation and its content, and is supposed to be merged to `feature/react-docs` but not until `feature/react-docs-develop` is merged to `feature/react-docs`.**

This PR adds the conditional blocks for every example container that has been transformed by the unfinished React transformer (**138** examples in total).

The current status of all the implemented React examples is tracked in this document: https://docs.google.com/spreadsheets/d/1Hk9m-O0VYWoSHGz-ib_x0biDXUgZmzXYtPBDMSfuxkQ.

For convenience, the table below lists the examples that still need some work (the list might not be complete, as there hasn't been thorough testing being done yet):

Status | File name  |  Problem
| :---: | -------------  | ------------- |
| ✅ | `internationalization/layout-direction.md` | The example container required passing the correct "preset" with all the needed dependencies.
| ✅  | `getting-started/saving-data.md`  | Needs work on event handling and output logging.
| ✅  | `getting-started/grid-size.md`  | Needs to be written manually (the transformer failed to generate the React snippet automatically).
| 🔲  | `getting-started/events-and-hooks.md`  | Needs to be written manually (the transformer failed to generate the React snippet automatically).<br>Also, there's an additional example container on the page (probably moved from the previously-existing React section). There might be some overlap between the examples' content - needs to be checked.
| ✅ | `formulas/formula-calculation.md` (2 examples) | Required changes in the `fixer.js` file and some minor fixes in the example container.
| 🔲  | `formulas/formula-calculation.md` (1 example) | Requires rewriting the form submission logic (possibly getting rid of the form?) + changing the "class" and "style" props to their React counterparts.
| 🔲  | `columns/column-summary.md` | The vanilla version throws the same errors as the React one, but for some reason the React version doesn't render at all - needs to be investigated.
| 🔲  | `columns/column-filter.md` |  Needs work on event handling.
| ✅ | `cell-types/numeric-cell-type.md` | The example container required passing the correct "preset" with all the needed dependencies.
| ✅ | `cell-types/numeric-cell-type.md` | There's an additional example container on the page (probably moved from the previously-existing React section). There might be some overlap between the examples' content - needs to be checked.
| ❓ | `cell-types/autocomplete-cell-type.md` | The example's not working properly - might be because of the AJAX nature of the example, Vanilla example seems to fail the same way.
| ✅ | `cell-functions/cell-renderer.md` | Additional react example already on the page (the 2nd example is actually the 3rd). Probably moved from the previously-existing React section, there might be some overlap;<br><br> The images are not rendered correctly for both Vanilla and React versions.
| ✅ | `cell-functions/cell-renderer.md` | Needs work on event handling.
| ✅ | `cell-features/selection.md` | Required minor changes in the container setup and the examples' content.
| ✅ | `cell-features/selection.md` | Original example changed, the transformed example needs to be unified with the updated example.
| ✅ | `cell-features/formatting-cells.md` | Required minor changes in the container setup.
| ✅ | `cell-features/clipboard.md` | Needs work on event handling.
| ✅ | `advanced-topics/batch-operations.md` | Needs work on event handling.
| ✅ | `accessories-and-menus/searching-values.md` | Needs work on event handling.

#### Additional notes:
1.  Non-example snippets are still written in vanilla JS, which makes no sense logically. At many places in the docs, there are step-by-step instructions (which at this point are still written in vanilla JS) that lead up to the last step being an example in React. 
2. There is a _HUGE_ number of snippets that needs to be manually rewritten.
3. In many instances there are too many auto-generated imports (like `useEffect`, when it's not being used in the example). That needs to be cleaned up.

### How has this been tested?
At the time of writing this, the `feature/react-docs-develop` is not finished, so all the tests were done by moving the documentation content to a clone of the `feature/react-docs` branch and run from there.

### Related issue(s):
1. #9694 

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

[skip changelog]

### CODE REVIEW

**Comment by warpech** [comment]
I made further, complete updates of the content of this branch with the previous PRs #9593 and #9576. My work is pushed to the branch `feature/issue-9694-verified-with-9593-and-9576`. 

@jansiegel after your review, I think it can be merged here. Please note my commit messages, especially the ones starting with `(warning)`.

**Comment by jansiegel** [comment]
> @jansiegel after your review, I think it can be merged here. Please note my commit messages, especially the ones starting with `(warning)`.

@warpech Cool! Could you make a PR to this branch, so I can review it?

**Comment by jansiegel** [comment]
@warpech Many of the examples seem to be fixed with today's PRs, there seem to be 5 issues left to correct (as far as I found).

[As discussed](https://handsoncode.slack.com/archives/C034WPQUG9W/p1659430751192409), I moved the notes to the markdown files as TODO entries, with `[react-docs]` annotation in https://github.com/handsontable/handsontable/pull/9695/commits/2ad4852ea8586fa7d94baef2348aa8818901cd41.

File: docs/.vuepress/public/scripts/fixer.js [hunk 1]
```diff
@@ -86,6 +86,9 @@
       } else if (key === 'handsontable/base') {
         ns = 'Handsontable';
 
+      } else if (key === 'hyperformula') {
+        ns = 'HyperFormula';
+
```
Comment (warpech) [inline docs/.vuepress/public/scripts/fixer.js:91]: What does this fix? How can I observe it?

Comment (jansiegel) [reply docs/.vuepress/public/scripts/fixer.js:91]: This is meant to enable our JSFiddle examples to simulate `import { HyperFormula } from 'hyperformula';` while using the global `HyperFormula`.

**Review by warpech — APPROVED** [review]
Congrats @jansiegel, it is a tremendous amount of work. 

I reviewed all of the changed pages and clicked through the demos. You've captured all of the remaining problems with the examples that I could find in my smoke test.

**Pull request merged**